client/java/generator/src/main/java/io/openlineage/client/JavaPoetGenerator.java

```diff
@@ -440,26 +529,84 @@ private void generateDefaultImplementation(
       containerTypeBuilder.addType(classBuilder.build());
 
 
-      Builder factory = MethodSpec.methodBuilder("new" + type.getName())
-          .addModifiers(PUBLIC)
-          .returns(getTypeName(type));
 
-      List<CodeBlock> factoryParams = new ArrayList<>();
 
-      type.getProperties().stream().filter(f -> !isASchemaUrlField(f)).forEach(f -> {
-        if (isAProducerField(f)) {
-          factoryParams.add(CodeBlock.of("this.producer"));
-        } else {
-          factory.addParameter(ParameterSpec.builder(getTypeName(f.getType()), f.getName()).build());
-          factory.addJavadoc("@param $N $N\n", f.getName(), f.getDescription() == null ? "the " + f.getName() : f.getDescription());
-          factoryParams.add(CodeBlock.of("$N", f.getName()));
-        }
-      });
-      factory.addJavadoc("@return $N", type.getName());
-      factory.addCode("return new $N(", "Default" + type.getName());
-      factory.addCode(CodeBlock.join(factoryParams, ", "));
-      factory.addCode(");\n");
-      containerTypeBuilder.addMethod(factory.build());
+      if (type.getProperties().stream().anyMatch(f -> isADeletedField(f))) {
+        // allow creating a Deleted Facet as well
+
+        // factory method
+        Builder factory = MethodSpec.methodBuilder("newDeleted" + type.getName())
+            .addModifiers(PUBLIC)
+            .returns(getTypeName(type));
+
+        List<CodeBlock> factoryParams = new ArrayList<>();
+
+        handleProperties(type, new ResolvedFieldHandler() {
+
+          @Override
+          public void onProducer(ResolvedField f) {
+            // the producer is automatically set
+            factoryParams.add(CodeBlock.of("this.producer"));
+          }
+
+          @Override
+          public void onDeleted(ResolvedField f) {
+            // this is the case where we create a deleted facet
+            factoryParams.add(CodeBlock.of("true"));
+          }
+
+          @Override
+          public void onField(ResolvedField f) {
+            factory.addParameter(ParameterSpec.builder(getTypeName(f.getType()), f.getName()).build());
+            factory.addJavadoc("@param $N $N\n", f.getName(), f.getDescription() == null ? "the " + f.getName() : f.getDescription());
+            factoryParams.add(CodeBlock.of("$N", f.getName()));
+          }
+
+        });
+
+        factory.addJavadoc("@return a deleted $N", type.getName());
+        factory.addCode("return new $N(", "Default" + type.getName());
+        factory.addCode(CodeBlock.join(factoryParams, ", "));
+        factory.addCode(");\n");
+        containerTypeBuilder.addMethod(factory.build());
+      }
```

Comment:
isn't the `else` missing here?

Reply:
Ah I see how this is misleading. I'll move things around a bit to make the code clearer.
Short answer: `else` is not missing here.
The code bellow this comment is the existing code that I have modified to use my new `handleProperties` method that forces me to handle all cases of special properties (schemaURL, deleted, producer)
I have added a scoping `{ }` block. All this does is make sure the variable declared in the block are not visible outside. That's because I added the block above that creates `newDeletedDatasetFacet` exists only for those classes with the deleted field.
It is actually not necessary after i moved things around. I'll clean that up.